bookwyrm/templates/lists/list.html

```diff
@@ -13,10 +13,51 @@
+                {% elif from_start == 3 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:-3 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:-3 }}</a>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:-2 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:-2 }}</a>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:-1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:-1 }}</a>
+                {% elif from_start > 3 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page=1&direction={{ direction }}&sort_by={{ sort_by }}">1</a>
+                    <span class="pagination-ellipsis">&hellip;</span>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:-1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:-1 }}</a>
+                {% endif %}
+            <a class="pagination-link is-current" href="{% url 'list' list.id %}?page={{ page }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page }}</a>
+                {% if from_end == 1 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:1 }}</a>
+                {% elif from_end == 2 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:1 }}</a>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:2 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:2 }}</a>
+                {% elif from_end == 3 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:1 }}</a>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:2 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:2 }}</a>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:3 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:3 }}</a>
+                {% elif from_end > 3 %}
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page|add:1 }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page|add:1 }}</a>
+                    <span class="pagination-ellipsis">&hellip;</span>
+                    <a class="pagination-link" href="{% url 'list' list.id %}?page={{ page_count }}&direction={{ direction }}&sort_by={{ sort_by }}">{{ page_count }}</a>
+                {% endif %}
+            </ul>
+        </nav>
+        {% endif %}
```

Comment:
I just caught on to what you meant by pagination snippet. https://github.com/mouse-reeve/bookwyrm/blob/main/bookwyrm/templates/snippets/pagination.html

I kinda like having the pagination list items for adjacent pages beyond Next and Previous. What do you think of me creating an alternative snippet for the list pagination that implements those?

Reply:
oh sorry I should have linked to that! And that's fine by me, there's a design pattern for that in bulma as well: https://bulma.io/documentation/components/pagination/

Reply:
Ok. Added a new snippet for the list. I think this will be easier in Django 3.2 https://docs.djangoproject.com/en/3.2/ref/paginator/#django.core.paginator.Paginator.get_elided_page_range